File: alt_raw_compilers/compiler.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"os/exec"
	"strings"
)

func clearConsole() {
	cmd := exec.Command("cmd", "/c", "cls")
	cmd.Stdout = os.Stdout
	cmd.Run()
}

func processLine(line string) string {
	if strings.Contains(line, "original_input") {
		return "solidity_output"
	} // else if (...)
	return line
}

func main() {
	clearConsole()

	if len(os.Args) < 2 {
		fmt.Println("Please provide a valid Solidity+ source file.")
		os.Exit(1)
	}

	inputFilePath := os.Args[1]
	outputFilePath := strings.TrimSuffix(inputFilePath, ".sol") + "_compiled.sol"

	// Check if input file exists
	if _, err := os.Stat(inputFilePath); os.IsNotExist(err) {
		fmt.Printf("File not found: %s\n", inputFilePath)
		os.Exit(1)
	}

	// Delete any existing output file
	if _, err := os.Stat(outputFilePath); err == nil {
		os.Remove(outputFilePath)
	}

	inputFile, err := os.Open(inputFilePath)
	if err != nil {
		fmt.Println("Error opening input file:", err)
		os.Exit(1)
	}
	defer inputFile.Close()

	outputFile, err := os.Create(outputFilePath)
	if err != nil {
		fmt.Println("Error creating output file:", err)
		os.Exit(1)
	}
	defer outputFile.Close()

	scanner := bufio.NewScanner(inputFile)
	writer := bufio.NewWriter(outputFile)

	for scanner.Scan() {
		line := scanner.Text()
		processedLine := processLine(line)
		writer.WriteString(processedLine + "\n")
	}

	if err := writer.Flush(); err != nil {
		fmt.Println("Error writing to output file:", err)
		os.Exit(1)
	}

	fmt.Printf("Compilation complete. Output file: %s\n", outputFilePath)
}

File: alt_raw_compilers/compiler.rs
use std::fs::{self, File};
use std::io::{self, BufRead, BufReader, Write};
use std::path::Path;
use std::process::Command;

fn clear_console() {
    Command::new("cmd")
        .args(&["/C", "cls"])
        .status()
        .unwrap();
}

fn process_line(line: &str) -> String {
    if line.contains("original_input") {
        "solidity_output".to_string()
    } // else if (...)
    else {
        line.to_string()
    }
}

fn main() -> io::Result<()> {
    clear_console();

    let args: Vec<String> = std::env::args().collect();
    if args.len() < 2 {
        println!("Please provide a valid Solidity+ source file.");
        std::process::exit(1);
    }

    let input_file_path = &args[1];
    let output_file_path = input_file_path.trim_end_matches(".sol").to_owned() + "_compiled.sol";

    if !Path::new(input_file_path).exists() {
        println!("File not found: {}", input_file_path);
        std::process::exit(1);
    }

    if Path::new(&output_file_path).exists() {
        fs::remove_file(&output_file_path)?;
    }

    let input_file = File::open(input_file_path)?;
    let reader = BufReader::new(input_file);

    let mut output_file = File::create(&output_file_path)?;

    for line in reader.lines() {
        let line = line?;
        let processed_line = process_line(&line);
        writeln!(output_file, "{}", processed_line)?;
    }

    println!("Compilation complete. Output file: {}", output_file_path);

    Ok(())
}
